compiler/lib/parse_bytecode.ml

```diff
@@ -1554,7 +1554,17 @@ and compile infos pc state instrs =
         let x, state = State.fresh_var state loc in
 
         if debug_parser () then Format.printf "%a = %a[%d]@." Var.print x Var.print y n;
-        compile infos (pc + 2) state ((Let (x, Field (y, n)), loc) :: instrs)
+        compile
+          infos
+          (pc + 2)
+          state
+          (( Let
+               ( x
+               , Prim
+                   ( Extern "caml_floatarray_unsafe_get"
```

Comment:
I wouldn't it be cleaner to extend the `Field` constructor  with some additional information ? Float | NonFloat ?

Reply:
That would be cleaner. I’m on it, it will take me a bit of work.

Reply:
Done